### ISSUE TITLE
settings for PAT or OAuth

### ISSUE THREAD

**Issue by mortenrs** [ankitbko/vscode-pull-request-azdo#81]
The new OAuth login do not work for me on self hosted Azure server.

Can you make a setting that select to use the PAT for login?

**Comment by ankitbko** [comment]
Sure I will add that as an option. What do you mean by self hosted server? You mean onpremise instance?

**Comment by mortenrs** [comment]
Yes you can call it onpremise. It requires VPN to access it.

Error log:
```
[Debug 1700924726.387s] AzdoRepository> Auth> Failed: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}

[Debug 1700924726.387s] Error: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}
	at _.<anonymous> (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:873741)
	at Generator.next (<anonymous>)
	at e (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:871091)
...

[Debug 1700925169.772s] AzdoRepository> Auth> Failed: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}

[Debug 1700925169.772s] Error: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}
	at _.<anonymous> (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:873741)
	at Generator.next (<anonymous>)
	at e (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:871091)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

[Debug 1700925169.773s] AzdoRepository> Auth> Failed: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}

[Debug 1700925169.773s] Error: ﻿{"$id":"1","innerException":null,"message":"TF400813: Resource not available for anonymous access. Client authentication required.","typeName":"Microsoft.TeamFoundation.Framework.Server.UnauthorizedRequestException, Microsoft.TeamFoundation.Framework.Server","typeKey":"UnauthorizedRequestException","errorCode":0,"eventId":3000}
	at _.<anonymous> (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:873741)
	at Generator.next (<anonymous>)
	at e (c:\Users\ <hidden> \.vscode\extensions\ankitbko.vscode-pull-request-azdo-0.2.1\dist\extension.js:5:871091)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

[Debug 1700925169.777s] Review> Validate state in progress

```

**Comment by toquanghieu** [comment]
same my issue, could we add an option in ext setting to choose auth method

**Comment by fabianem** [comment]
same for me, that option for the PAT would be great

**Comment by maxmitti** [comment]
Any update on this?
I also have issues with on on-premise instance.
I was able to login with a PAT token using version 0.0.25 of this extension, but I can’t find a way to use a PAT token with version 0.2.1.

**Comment by ankitbko** [comment]
Fixed in v0.2.2 https://github.com/ankitbko/vscode-pull-request-azdo/pull/83
